presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriterFactory.java

```diff
@@ -384,4 +395,18 @@ private boolean isMapStatisticsEnabled(Properties schema)
     {
         return parseBoolean(schema.getProperty(ORC_MAP_STATISTICS_KEY, "false"));
     }
+
+    // The result is cached by the Suppliers.memoize because getCanonicalHostName does a DNS call.
+    // Guava's MemoizingSupplier produced by the Suppliers.memoize does internal synchronization on get(),
+    // that's why we don't need to synchronize this method.
+    private static Optional<String> getHostname()
+    {
+        try {
+            String canonicalHostname = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase(Locale.US);
+            return Optional.of(canonicalHostname);
```

Comment:
noob question. Can't we pass down the host from the caller instead of using the `InetAddress` approach.
A quick search for caller for `createFileWriter` shows `HiveWriterFactory`
https://github.com/prestodb/presto/blob/3fbbb677da4c9dc2386379d6a9d523fc2ae3c1c8/presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java#L371-L380

`HiveWriterFactory` has the current node information from `nodeManager.getCurrentNode().getHost()`

Reply:
> Can't we pass down the host from the caller instead of using the InetAddress approach.

We can, but it will impact all file formats, when I only wanted to change it fro ORC. For example Alpha already writes the hostname on the native side.

> HiveWriterFactory has the current node information from nodeManager.getCurrentNode().getHost()
I started with with the nodeManager, but it turns out that it returns the IP address, which is more dynamic that the hostname.